aff4/aff4_map.cc

```diff
@@ -79,13 +81,19 @@ AFF4Status AFF4Map::LoadFromURN() {
             // Clear the map so we start with a fresh map.
             Clear();
         } else {
-            // Note: The legacy AFF4 version, and AFF4 Standard use the same map format.
-            BinaryRange binary_range;
-            while (map_stream->ReadIntoBuffer(
-                        &binary_range, sizeof(binary_range)) == sizeof(binary_range)) {
-                Range range(binary_range);
+            // Calculate number of ranges
+            const auto n = map_stream->Size() / sizeof(BinaryRange);
+            auto buffer = new BinaryRange[n];
```

Comment:
maybe use a uniq_ptr here?

Reply:
Originally I wanted to use a unique_ptr, but the variant of unique_ptr that handles array style pointers is a C++14 feature and the project only requires C++11.

Reply:
The memory doesn't leave the scope of the block before being deleted anyway, so the benefits of unique_ptr are minimal, and probably not worth requiring C++14 just for this.

Reply:
I didnt know that  unique_ptr<BinaryRaneg[]> buffer(new BinaryRange[n]());  is a c++14 feature.

This suggests it is c++11:
http://www.cplusplus.com/reference/memory/unique_ptr/operator[]/

Reply:
It might have actually just been the std::make_unique(size_t) call that I was using that's C++14, but again, not a whole lot of reason to use the smart pointer here.